segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataCleaner.java

```diff
@@ -180,24 +178,24 @@ protected void doStop() {
                 }, this.executor);
     }
 
-    private CompletableFuture<Void> deleteUnusedTransientSegments() {
-        long lastSeqNo = this.lastIterationSequenceNumber.getAndSet(this.metadata.getOperationSequenceNumber());
+    private CompletableFuture<Void> deleteUnusedTransientSegments(long lastSeqNo) {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.traceObjectId, "deleteUnusedTransientSegments", lastSeqNo);
-
         // Serialize only those segments that are still alive (not deleted or merged - those will get removed anyway).
         this.metadata.getEvictionCandidates(lastSeqNo, this.config.getMaxConcurrentSegmentEvictionCount())
                 .stream()
-                .filter(sm -> sm.getType() == SegmentType.TRANSIENT_SEGMENT && sm.isDeleted())
+                .filter(sm -> sm.getType().isTransientSegment() && !sm.isDeleted())
+                //.filter(sm -> appendProcessor.isTransientSegmentActive(sm.getName()))
                 .forEach(sm -> {
-                    // Todo: AppendProcessor - isTransientSegmentActive()?
+                    metadataStore.deleteSegment(sm.getName(), TIMEOUT);
```

Comment:
This is not awaiting the completion. You are just initiating these delete operations.